### PR TITLE
Add `distributed_port` for `deepspeed.initialize`

### PR DESCRIPTION
**Pull request by LZHgrla**
`deepspeed.initialize` does not involve the `distributed_port` argument, and always uses `TORCH_DISTRIBUTED_DEFAULT_PORT` to initialize the dist env

### CODE REVIEW

**Review by tjruwase — APPROVED** [review]

**Comment by LZHgrla** [comment]
@tjruwase Hi!
I merged the main branch into this PR. Waiting for the approval of the CI.

**Comment by LZHgrla** [comment]
@tjruwase 
Hi! Any update about this PR?

**Pull request merged**
